src/coreclr/src/jit/morph.cpp

```diff
@@ -6094,6 +6097,10 @@ GenTree* Compiler::fgMorphField(GenTree* tree, MorphAddrContext* mac)
             {
                 offsetNode = gtNewIndOfIconHandleNode(TYP_I_IMPL, (size_t)tree->AsField()->gtFieldLookup.addr,
                                                       GTF_ICON_FIELD_HDL, false);
+#ifdef DEBUG
+                offsetNode->gtGetOp1()->AsIntCon()->gtMethodHandle =
+                    (CORINFO_METHOD_HANDLE)GenTreeIntCon::MethodHandleType::FieldAccess;
```

Comment:
Is this a place where we should save `gtFldObj`?

Reply:
Correct.